retrofit2-clients/src/main/java/com/palantir/remoting2/retrofit2/Retrofit2ClientBuilder.java

```diff
@@ -40,25 +37,20 @@
 import retrofit2.Retrofit;
 import retrofit2.converter.jackson.JacksonConverterFactory;
 
-public final class Retrofit2ClientBuilder extends ClientBuilder {
+public final class Retrofit2ClientBuilder {
 
     private static final ObjectMapper CBOR_OBJECT_MAPPER = ObjectMappers.newCborClientObjectMapper();
     private static final ObjectMapper OBJECT_MAPPER = ObjectMappers.newClientObjectMapper();
 
-    private final ClientConfig config;
+    private final ClientConfiguration config;
 
-    public Retrofit2ClientBuilder() {
-        this.config = ClientConfig.builder().build();
-    }
-
-    public Retrofit2ClientBuilder(ClientConfig config) {
+    public Retrofit2ClientBuilder(ClientConfiguration config) {
+        Preconditions.checkArgument(!config.uris().isEmpty(), "Cannot construct retrofit client with empty URI list");
```

Comment:
it's needed here since otherwise the failover thing cannot be configured. it's a limitation of how we can interfere with the underlying retrofit thing.